bin/all_sky_search/pycbc_template_kde_calc

```diff
@@ -58,51 +61,57 @@ def optimizedparam(sampleval, nfold=2):
 if args.make_signal_kde and args.make_template_kde:
     raise Exception("Choose only one option out of --make-signal-kde and --make-template-kde")
 
-#calling template data and constructing template_kde_hdf file over template params
-template_file = h5py.File(args.template_file, 'r')
-mass1_n = template_file['mass1'][:]
-mass2_n = template_file['mass2'][:]
-spin1z_n = template_file['spin1z'][:]
-spin2z_n = template_file['spin2z'][:]
-chi_eff_n = convert.chi_eff(mass1_n, mass2_n, spin1z_n, spin2z_n)
-mchirp_n = convert.mchirp_from_mass1_mass2(mass1_n, mass2_n)
-eta_n = convert.eta_from_mass1_mass2(mass1_n, mass2_n)
+#calling template data and constructing template_kde_hdf_file over same template params
+temp_file = h5py.File(args.template_file, 'r')
+approximant = temp_file['approximant'][:]
+f_lower = temp_file['f_lower'][:]
+tid = numpy.arange(len(f_lower))
+mass1, mass2, spin1z, spin2z = triggers.get_mass_spin(temp_file, tid)
+chi_eff = convert.chi_eff(mass1, mass2, spin1z, spin2z)
+mchirp = convert.mchirp_from_mass1_mass2(mass1, mass2)
+eta = convert.eta_from_mass1_mass2(mass1, mass2)
 
 assert len(args.fit_param) == len(args.log_param)
+parvals = []
 
-if args.make_template_kde:
-    param0 = eval(args.fit_param[0][:])
-    param1 = eval(args.fit_param[1][:])
-    param2 = eval(args.fit_param[2][:])
-    if args.log_param == ['False', 'False', 'False']:
-        sample_n = numpy.vstack((param0, param1, param2)).T
-    elif args.log_param == ['True', 'False', 'False']:
-        sample_n = numpy.vstack((numpy.log(param0), param1, param2)).T
-    elif args.log_param == ['True', 'True', 'False']:
-        sample_n = numpy.vstack((numpy.log(param0), numpy.log(param1), param2)).T
+for param, slog in zip(args.fit_param, args.log_param):
+    data = triggers.get_param(param, args, mass1, mass2, spin1z, spin2z)
+    if slog in ['false', 'False', 'FALSE']:
+        logging.info('Using param: %s', param)
+        parvals.append(data)
+    elif slog in ['true', 'True', 'TRUE']:
+        logging.info('Using log param: %s', param)
+        parvals.append(numpy.log(data))
     else:
-        raise ValueError("invalid log_param argument, use 'True', or 'False'")    
+        raise ValueError("invalid log param argument, use 'true', or 'false'")
```

Comment:
this should be True and False, exactly as the choices you allow in the argparse setup